website/routes.py

```diff
@@ -1137,12 +1137,12 @@ def make_url_map(app):
 
         # Tags
         Rule([
-            '/project/<pid>/addtag/<tag>/',
-            '/project/<pid>/node/<nid>/addtag/<tag>/',
+            '/project/<pid>/addtag/',
```

Comment:
We try to avoid routes with verbs in them, as they are redundant with the HTTP verbs. `/project/<pid>/tags/` is better than `project/<pid>/addtag/`. 

Also, for backwards-compatibility, it would be good to redirect the old route to the new routes.